components/esp_modem/test/target_ota/components/manual_ota/transport_batch_tls.cpp

```diff
@@ -0,0 +1,232 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+#include <vector>
+#include "esp_log.h"
+#include "mbedtls_wrap.hpp"
+#include "esp_transport_tcp.h"
+
+#define TAG "batch-tls"
+
+class TlsTransport: public Tls {
+public:
+    explicit TlsTransport(esp_transport_handle_t parent) : Tls(), transport_(parent), read_len(0), offset(0) {}
```

Comment:
NItpick: last timeout is not initialized.

Reply:
will init